packages/firestore/test/unit/specs/spec_test_runner.ts

```diff
@@ -1004,8 +1011,10 @@ abstract class TestRunner {
     // without any direct user interaction.
     // TODO(multitab): Refactor so this is only executed after primary tab
     // change
-    if (!obj.isEmpty(this.expectedActiveTargets)) {
-      await this.connection.waitForWatchOpen();
+    if (this.expectedActiveTargets) {
+      if (!obj.isEmpty(this.expectedActiveTargets)) {
+        await this.connection.waitForWatchOpen();
+      }
       await this.queue.drain();
```

Comment:
Er, why do we need to call/wait-for queue.drain() even when we don't call waitForWatchOpen()?  validateActiveTargets() is called by validateStateExpectations() which is called by run(step) which already calls queue.drain()...